src/drivers/common/memory.c

```diff
@@ -100,4 +103,13 @@ static int periph_memory_init(void) {
 	return 0;
 }
 
+int periph_mem_size() {
+	return ARRAY_SPREAD_SIZE(__periph_mem_registry);
+}
+
+void periph_elem_desc(struct periph_memory_desc *elem, int num) {
```

Comment:
I think, that the problem was caused by devision into MMU_PAGE_SIZE, during _segments[] init: variables were round, so we couldn't restore correct len using them.

Fixed that by using __periph_mem_registry instead of _segments[] array.

Reply:
You are right, periph_desc corresponds to the memory region, which the corresponding peripheral device has requested, while _segments correspond to the _pages_ that will be allocated for these regions. I think now it's well done.

Reply:
By the way, it would be nice to print also the name of the corresponding memory segment. But let's defer this task for the future.